lib/msf/core/post/linux/kernel.rb

```diff
@@ -0,0 +1,58 @@
+# -*- coding: binary -*-
+require 'msf/core/post/common'
+
+module Msf
+class Post
+module Linux
+module Kernel
+  include ::Msf::Post::Common
+
+  #
+  # Returns the kernel version
+  #
+  # @return [String]
+  #
+  def kernel_version
+    cmd_exec('uname -r').to_s
+  rescue
+    raise 'Could not determine kernel version'
+  end
+
+  #
+  # Returns true if kernel and hardware supports Supervisor Mode Access Prevention (SMAP), false if not.
+  #
+  # @return [Boolean]
+  #
+  def smap_enabled?
+    cmd_exec('cat /proc/cpuinfo').to_s.include? 'smap'
+  rescue
+    raise 'Could not determine SMAP status'
+  end
+
+  #
+  # Returns true if kernel and hardware supports Supervisor Mode Execution Protection (SMEP), false if not.
+  #
+  # @return [Boolean]
+  #
+  def smep_enabled?
+    cmd_exec('cat /proc/cpuinfo').to_s.include? 'smep'
+  rescue
+    raise 'Could not determine SMEP status'
+  end
+
```

Comment:
I've looked for shem and sham before
https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/linux/local/netfilter_priv_esc_ipv4.rb#L91

Reply:
Good idea

Reply:
Uh wtf is `shem` / `sham` and why is the netfilter exploit the only reference I can find?

Reply:
https://www.youtube.com/watch?v=QwRISkyV_B8

Reply:
My research took me in a... different direction.

![Shem / Sham](https://user-images.githubusercontent.com/434827/38463711-0fb80858-3b44-11e8-90ac-70128e3a9d51.png)